src/main/java/org/zalando/putittorest/HttpClientFactoryBean.java

```diff
@@ -36,9 +40,17 @@ public void setSocketTimeout(final int socketTimeout) {
         config.setSocketTimeout(socketTimeout);
     }
 
+    @Hack("Hide type to avoid runtime dependency")
+    public void setMetricsWrapper(final Object metricsWrapper) {
+        this.metricsWrapper = metricsWrapper;
+    }
+
     @Override
     public CloseableHttpClient getObject() {
         builder.setConnectionTimeToLive(30, TimeUnit.SECONDS);
+        if (metricsWrapper != null) {
+            builder.setRequestExecutor(new InstrumentedHttpRequestExecutor((MetricsWrapper) metricsWrapper));
```

Comment:
Ease of implementation and because today I am sure that there is no way for a client to pass in her own executor.

As this might change I will try to get a solution with two interceptors into `zmon-actuator` and adapt this PR.